portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -2339,5 +2349,33 @@ RenderResult getInlineRenderingForTool(HttpServletResponse res, HttpServletReque
 		
 		return rval;
 	}
+	
+	/**
+	 * Checks if current user is being impersonated (via become user/sutool) and returns displayId of
+	 * the impersonator. Adapted from SkinnableLogin's isImpersonating()
+	 * @return displayId of impersonator, or empty string if not being impersonated
+	 */
+	private String getImpersonatorDisplayId()
+	{
+		Session currentSession = SessionManager.getCurrentSession();
+		String currentUserId = currentSession.getUserId();
+		UsageSession originalSession = (UsageSession) currentSession.getAttribute(UsageSessionService.USAGE_SESSION_KEY);
+		String originalUserId = originalSession == null ? "" : originalSession.getUserId();
+		
+		if (originalSession != null && !originalUserId.equals(currentUserId))
```

Comment:
I recall now, the check is here to avoid false positive caused by setting the originalUserId to empty string above. I think I can refactor to do the null check first.